### PR TITLE
Always clobber existing outputs in FERC to SQLite conversions

### PR DESCRIPTION
**Pull request by zaneselvans**
# Overview

Always clobber the FERC SQLite DBs when re-running the `ferc_to_sqlite` jobs, and remove the `--clobber` option from the CLI. Partly in response to confusion that came up in #3612 

# Testing

* With existing FERC databases (and JSON outputs) I re-ran both the fast and full FERC to SQLite jobs in the Dagster UI.
* Also checked that the CLI using `ferc_to_sqlite src/pudl/package_data/settings/etl_full.yml` with existing outputs, and it ran with no complaints.

```[tasklist]
# To-do list
- [x] Update the release notes
- [x] Review the PR yourself and call out any questions or issues you have
- [x] Ensure docs build, unit & integration tests, and test coverage pass locally with `make pytest-coverage`
```

### CODE REVIEW

File: docs/dev/run_the_etl.rst [hunk 1]
```diff
@@ -222,19 +222,7 @@ the job. The bottom part of the window contains dagster logs. You can
 view logs from the ``pudl`` package in the CLI window the ``dagster-webserver`` process
 is running in.
 
-If you need to set op configurations, such as the ``clobber`` setting, you can
-add them in the Launchpad tab of the job like so::
```
Comment (zaneselvans) [inline docs/dev/run_the_etl.rst:226]: Is there another example that we should put in here instead?

File: docs/dev/run_the_etl.rst [hunk 2]
```diff
@@ -471,15 +459,8 @@ what the script is doing, and file outputs the directory you specified via the
...
 
-If you need to re-run ``ferc_to_sqlite`` and want to overwrite their previous outputs
-you can add ``--clobber`` (run ``ferc_to_sqlite --clobber``).  All of the PUDL scripts
-also have help messages if you want additional information (run ``script_name --help``).
-
-.. note::
-
-  The ``pudl_etl`` command does not have a ``--clobber`` option because each etl run
-  uses the same database file to read and write tables.  This facilitates re-running
-  small portions of the ETL using Dagster.
```
Comment (zaneselvans) [inline docs/dev/run_the_etl.rst:482]: This no longer seemed necessary once the other discussion of clobber was gone.

File: test/unit/extract/xbrl_test.py [hunk 3]
```diff
@@ -135,68 +134,6 @@ def test_xbrl2sqlite(settings, forms, mocker, tmp_path):
...
-                xbrl_num_workers=10,
-            ),
-        },
-    )
-
-    assert ferc1_sqlite_path.exists()
-    xbrl2sqlite_op_factory(XbrlFormNumber.FORM1)(context)
-    assert not ferc1_sqlite_path.exists()
-
-
```
Comment (zaneselvans) [inline test/unit/extract/xbrl_test.py:199]: My interpretation of these is tests is that they were only checking the clobber/no-clobber behavior, which should no longer be necessary.

File: docs/dev/dev_setup.rst [hunk 4]
```diff
@@ -44,6 +44,13 @@ with the following commands:
     $ mamba update mamba
     $ conda config --set channel_priority strict
 
+.. note::
+
+   Make sure that after you've installed ``conda`` or ``mamba`` that you can activate
+   and deactivate the ``base`` environment (which is created by default). Try creating
+   a simple new environment and activating it to make sure everything is working and
+   your shell has been correctly configured to use the new package manager.
+
```
Comment (zaneselvans) [inline docs/dev/dev_setup.rst:53]: @denimalpaca would this have helped avoid the conda environment initialization problem you ran into?

File: test/unit/extract/xbrl_test.py (hunk 3 above)
Comment (jdangerx) [reply test/unit/extract/xbrl_test.py:199]: Yeah, and they also occasionally screwed up nightly builds for complicated reasons. 🔥

File: docs/dev/dev_setup.rst [hunk 5]
```diff
@@ -97,11 +104,12 @@ Makefiles work, check out `this excellent Makefile tutorial
 <https://makefiletutorial.com/>`__
 
 To create the ``pudl-dev`` environment and install the local PUDL package using
-``make``, run:
+``make``, and then activate the newly created environment run:
```
Comment (jdangerx) [inline docs/dev/dev_setup.rst:107]: Non-blocking: something like this reads a little better to me.

```
To create the ``pudl-dev`` environment, install the local PUDL package, and activate the newly created environment, run:
```

File: docs/dev/dev_setup.rst [hunk 6]
```diff
@@ -245,7 +253,10 @@ stuff should go. We call this a "PUDL workspace".
 
 First, create a directory to store local caches of raw PUDL data. You can put this
 anywhere, but we put this in ``~/pudl_input`` in the documentation.  Then create an
-environment variable called ``PUDL_INPUT`` to store the path to this new directory:
+environment variable called ``PUDL_INPUT`` to store the path to this new directory and
+make sure that it is set whenever you start up a new shell. These shorthand commands
+will append a line to the end of your shell initialization file. If you need to change
+it later you'll want to edit those files directly:
```
Comment (jdangerx) [inline docs/dev/dev_setup.rst:259]: Non-blocking: while we're here, we can tell them 

```
Please replace ``/absolute/path/to/pudl_input`` with the actual path to the directory.
```

I've answered several environment questions from people who've copy-pasted without noticing that that's a fake path.

File: docs/dev/run_the_etl.rst (hunk 1 above)
Comment (jdangerx) [reply docs/dev/run_the_etl.rst:226]: I feel like the only times I ever used this were for `clobber` - nothing comes to mind.

File: docs/release_notes.rst [hunk 7]
```diff
@@ -126,6 +126,13 @@ Bug Fixes
   used on any dataframe, with every column encoded based on the field definitions and
   FK constraints associated with the column name.
 
+CLI Changes
+^^^^^^^^^^^
+* Removed the ``--clobber`` option from the ``ferc_to_sqlite`` command and associated
+  assets. We rebuild these databases infrequently, and needing either edit the runtime
```
Comment (jdangerx) [inline docs/release_notes.rst:132]:
```suggestion
  assets. We rebuild these databases infrequently, and needing to either edit the runtime
```

**Review by jdangerx — APPROVED** [review]
Seems good, just a couple small suggestions for the docs that are non blocking.

**Pull request merged**